src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -434,14 +434,24 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state, GCHand
                 // Set maximum automatic redirection option
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_MAXREDIRS, _maxAutomaticRedirections);
             }
-            if (state.RequestMessage.Content != null)
+
+            if (state.RequestMessage.Method == HttpMethod.Put)
             {
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_UPLOAD, 1L);
             }
-            if (state.RequestMessage.Method == HttpMethod.Head)
+            else if (state.RequestMessage.Method == HttpMethod.Head)
             {
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_NOBODY, 1L);
             }
+            else if (state.RequestMessage.Method == HttpMethod.Post)
```

Comment:
I think that's fine, especially if we're eventually going to have more conditions here. I personally like:
```C#
switch (state.RequestMessage.Method)
{
    case HttpMethod.Put:
       ...
    case HttpMethod.Head:
        ...
    case HttpMethod.Post:
        ...
}
```
better than:
```C#
if (state.RequestMessage.Method == HttpMethod.Put)
{
    ...
}
else if (state.RequestMessage.Method == HttpMethod.Head)
{
    ...
}
else if (state.RequestMessage.Method == HttpMethod.Post)
{
    ...
}
```
but it's not a big deal.  Whichever you prefer.